.github/workflows/ci.yml

```
@@ -0,0 +1,11 @@
+name: CI
+
+on: push
```

You probably want
```suggestion
on:
  push:
    tags: ["*"]
    branches: [main]
  pull_request:
  workflow_dispatch:
```

This will create a PR CI, but also build commits that land in main. The workflow_dispatch is also extremely handy as it lets you manually run the Workflow for a specific branch